Makefile

```diff
@@ -90,13 +90,24 @@ helm-template:
 
 default: helm-template
 
+generate-s2i-docs:
+	./docs/scripts/generate-s2i-docs.sh
+
+clean-s2i-docs:
+	rm docs/s2i-*
+
 # renders all "task-*" named templates into the release directory, using the original
 # template filename as the Task name on release direcotry.
 helm-template-tasks:
 	mkdir -p $(RELEASE_DIR) || true
-	for t in `ls -1 templates/task-*.yaml`; do \
-		helm template --show-only=$$t . >$(RELEASE_DIR)/`basename $$t`; \
+	for t in templates/task-*.yaml; do \
+	    task_file="$$(basename $${t})"; \
+	    echo "# Rendering template $${t} into $${task_file}"; \
+	    make helm-template ARGS="--show-only=$${t}" >${RELEASE_DIR}/$${task_file}; \
 	done
+	make generate-s2i-docs
+	find docs -maxdepth 1 -type f -exec cp {} $(RELEASE_DIR) \;
+	make clean-s2i-docs
```

Comment:
Please introduce a `release` target (`make release`) to render the tasks and documentations files into a single folder (`${RELEASE_DIR}`).

Also, `Makefile` targets needs to be chained instead of calling `make` recursively, please check the other targets as example.